coredb-operator/tests/integration_tests.rs

```diff
@@ -391,6 +411,13 @@ mod test {
             },
             "spec": {
                 "replicas": 1,
+                "serviceAccountTemplate": {
```

Comment:
I would just only specify the serviceAccountTemplate on the last one here, then also add an assertion that the service account is created, it's connected to the pod, and the annotation look right on the service account